addons/mail/static/src/core/common/chat_window_model.js

```diff
@@ -21,7 +21,11 @@ export class ChatWindow extends Record {
     }
 
     actionsDisabled = false;
-    thread = Record.one("Thread");
+    thread = Record.one("Thread", {
+        onDelete() {
+            this.delete();
+        },
```

Comment:
It's probably better to call "close" to properly synchronize.

Note: when thread is deleted it might already be too late, it should probably be done in the delete flow instead.